core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java

```diff
@@ -182,9 +183,8 @@ public CompletionStage<Void> trackForStateTransfer(Collection<XSiteState> stateL
          int segment = topology.getSegment(state.key());
          IracManagerStateTransferState iracState = new IracManagerStateTransferState(segment, state.key());
          // if an update is in progress, we don't need to send the same value again.
-         if (updatedKeys.putIfAbsent(iracState.getKey(), iracState) == null) {
-            cf.dependsOn(iracState.getCompletionStage());
-         }
+         IracManagerKeyState existent = updatedKeys.computeIfAbsent(iracState.getKey(), ignore -> iracState);
+         cf.dependsOn(existent.subscribe());
```

Comment:
Found this during the `IracStateTransferTest#testConflict`. The test will disconnect the sites, put some values in each, and connect them again, waiting for the conflict resolution to take place. I found that:

1. The site will have some state to send to the backup, since the backup is offline we skip sending the command;
2. The manager invoked `IracResponseCollector#freeze`, which in turn calls the `IracResponseCollector#run`

Since the site was offline, no data was sent to the backup. But because the `IracResponseCollector` was created passing the batch of states and when `IracResponseCollector#run` executes it will not find anything in the collector's `failedKeys` set, and then call the `DefaultIracManager#onBatchResponse` saying that all states were successfully sent, and then clearing all states tracked. Even though nothing was sent to the backup because it was offline.

When the test calls for the state transfer, it will add the `IracManagerStateTransferState` to `updatedKeys` and will wait for it to complete/transfer. The test then is able to see that everything is transferred ok.

When I inserted my changes, we did not remove the elements from the `updatedKeys` because we were unable to send it to the backup because it was offline. So the state transfer can't insert the `IracManagerStateTransferState`, because the state is still tracked, which would cause to return saying that the state transfer was finished (the `iracExecutor.run()` in the next line is asynchronous) even though no state was transferred yet. I added the completable future to the `IracManagerKeyState` just so we are able to wait here for the transfer to complete.

Reply:
> When I inserted my changes, we did not remove the elements from the updatedKeys because we were unable to send it to the backup because it was offline.

This is a problem. Users can take a site offline for long periods and, in this case, you don't want to keep keys there wasting memory (and increasing memory consumption).